src/identityApiClient.ts

```diff
@@ -0,0 +1,257 @@
+import Constants, { HTTP_ACCEPTED, HTTP_OK } from './constants';
+import { FetchUploader, XHRUploader } from './uploaders';
+import { CACHE_HEADER } from './identity-utils';
+import { parseNumber } from './utils';
+import { IAliasCallback, IAliasRequest, IdentityAPIMethod, IIdentity, IIdentityAPIRequestData } from './identity.interfaces';
+import { IdentityApiData, MPID, UserIdentities } from '@mparticle/web-sdk';
+import { IdentityCallback, IdentityResultBody, IIdentityResponse } from './identity-user-interfaces';
+
+const {
+    HTTPCodes,
+    Messages,
+    IdentityMethods,
+} = Constants;
+
+const { Modify } = IdentityMethods;
+
+export interface IIdentityApiClient {
+    sendAliasRequest: (
+        aliasRequest: IAliasRequest,
+        aliasCallback: IAliasCallback
+    ) => Promise<void>;
+    sendIdentityRequest: (
+        identityApiRequest: IIdentityAPIRequestData,
+        method: IdentityAPIMethod,
+        callback: IdentityCallback,
+        originalIdentityApiData: IdentityApiData,
+        parseIdentityResponse: IIdentity['parseIdentityResponse'],
+        mpid: MPID,
+        knownIdentities: UserIdentities
+    ) => Promise<void>;
+    getUploadUrl: (method: IdentityAPIMethod, mpid: MPID) => string;
+    getIdentityResponseFromFetch: (
+        response: Response,
+        responseBody: IdentityResultBody
+    ) => IIdentityResponse;
+    getIdentityResponseFromXHR: (response: XMLHttpRequest) => IIdentityResponse;
+}
+
+
+export interface IAliasResponseBody {
+    message?: string
+}
+
+export default function IdentityAPIClient(this: IIdentityApiClient, mpInstance) {
+    this.sendAliasRequest = async function(aliasRequest, aliasCallback) {
```

Comment:
what's the best practice for when you define an interface like `IIdentityApiClient`...should everything in the signatures also be typed?  i expect then we don't need to add types, in which case we should remove them from the signature of `this.sendIdentityRequest`

whatever we pick, let's be consistent

Reply:
This is a good callout.

While I think we can avoid repeating ourselves because the IDE can infer the type based on the definition and signature, I can see a benefit to repeating the definition in that it puts the definition closer to the actual implementation.